packages/react-components/react-spinbutton/src/components/SpinButton/SpinButton.types.ts

```diff
@@ -91,46 +94,40 @@ export type SpinButtonCommons = {
    * - User *commits* edits to the input text by focusing away (blurring) or pressing enter.
    *   Note that this is NOT called for every key press while the user is editing.
    */
-  onChange: (event: SpinButtonChangeEvent, data: SpinButtonOnChangeData) => void;
+  onChange?: (event: SpinButtonChangeEvent, data: SpinButtonOnChangeData) => void;
 
   /**
    * How many decimal places the value should be rounded to.
    *
    * The default is calculated based on the precision of `step`: i.e. if step = 1, precision = 0.
    * step = 0.0089, precision = 4. step = 300, precision = 2. step = 23.00, precision = 2.
    */
-  precision: number;
+  precision?: number;
 
   /**
    * Controls the colors and borders of the input.
    * @default 'outline'
    */
-  appearance: 'outline' | 'underline' | 'filledDarker' | 'filledLighter';
+  appearance?: 'outline' | 'underline' | 'filledDarker' | 'filledLighter';
 
   /**
    * Size of the input.
    * @default 'medium'
    */
-  size: 'small' | 'medium';
+  size?: 'small' | 'medium';
 
   /**
    * Strings for localizing text in the control.
    */
   strings?: SpinButtonStrings;
 };
 
-/**
- * SpinButton Props
- */
-export type SpinButtonProps = Omit<ComponentProps<Partial<SpinButtonSlots>, 'input'>, 'onChange' | 'size'> &
-  Partial<SpinButtonCommons>;
-
 /**
  * State used in rendering SpinButton
  */
 export type SpinButtonState = ComponentState<SpinButtonSlots> &
-  Partial<SpinButtonCommons> &
-  Pick<SpinButtonCommons, 'appearance' | 'size'> & {
+  Partial<Exclude<SpinButtonProps, keyof SpinButtonSlots>> &
```

Comment:
I think this actually doesn't need to be added to state at all. It looks like `appearance` and `size` are the only props actually added to the state, and those are the only two that actually make sense to me.

Aside from that, if you keep this, I don't think `Partial` is needed, since all the props within it are already optional